bee-ternary/src/bigint/binary_representation.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 IOTA Stiftung
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and limitations under the License.
+
+//! Binary representation of big integers.
+
+use crate::bigint::sealed::Sealed;
+
+/// The number of bits in an U384/I384.
+pub const BINARY_LEN: usize = 384;
+
+/// Binary representation of a big integer.
+pub trait BinaryRepresentation: Sealed + Clone {
+    /// Inner representation type of the big integer.
+    type T;
+
+    /// Iterates over a slice of the inner representation type of the big integer.
+    fn iter(&self) -> std::slice::Iter<'_, Self::T>;
+}
+
+/// The number of u8s in an U384/I384.
+pub const BINARY_LEN_IN_U8: usize = BINARY_LEN / 8;
+
+/// The inner representation of an U384/I384 using 48 u8s.
+pub type U8Repr = [u8; BINARY_LEN_IN_U8];
+
+impl Sealed for U8Repr {}
```

Comment:
`BinaryRepresentation` requires `Sealed `, so you *have* to implement it.

Reply:
yea, but is it really necessary: Maybe I am wrong but it doesn't seem to be required, as [here](https://docs.serde.rs/src/serde_json/value/index.rs.html) they don't have to explicitly impl Sealed for the types they are implementing `Index` for. Never actually used that pattern myself, so I might be just confused.

Reply:
They do: https://docs.serde.rs/src/serde_json/value/index.rs.html#132

Reply:
meh, you're right. I didn't see it. Probably I wished, that Rust would autoimplement it, because with a statement like:
`impl BinaryRepresentation for u32` and `trait BinaryRepresentation: Sealed` I'ld assume Rust could do this if the trait is just a marker. All good then.